### PR TITLE
Add notes about the new CUDA version restrictions to the changelog an…

### PR DESCRIPTION
**Pull request by hawkinsp**
…d installation instructions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,18 +16,24 @@ Remember to align the itemized text with the first line of an item within a list
     * `random.rbg_key(seed)` becomes `random.PRNGKey(seed, impl='rbg')`
     * `random.unsafe_rbg_key(seed)` becomes `random.PRNGKey(seed, impl='unsafe_rbg')`
 * Changes:
-  * {func}`jax.scipy.stats.mode` now returns a 0 count if the mode is taken
-    across a size-0 axis, matching the behavior of `scipy.stats.mode` in SciPy
-    1.11.
+  * CUDA: JAX now verifies that the CUDA libraries it finds are at least as new
```
Comment (yashk2810) [inline CHANGELOG.md:19]: So this will break things if we ship cuda 12.2 and users have 12.1?

Comment (hawkinsp) [reply CHANGELOG.md:19]: Correct. JAX will raise an exception if you do this.

Comment (yashk2810) [reply CHANGELOG.md:19]: What was the reason? (sorry I don't have the context nor did I know about this)

Comment (hawkinsp) [reply CHANGELOG.md:19]: I'm fairly confident this is responsible for a number of user issue where too old a  version is chosen and mysterious failures happen. I'd much rather we caught this early and loudly.

Comment (yashk2810) [reply CHANGELOG.md:19]: Ok

**Review by yashk2810 — APPROVED** [review]

**Pull request merged**